pkg/utils/miscellaneous.go

```diff
@@ -71,6 +71,24 @@ func MergeStringMaps[T any](oldMap map[string]T, newMaps ...map[string]T) map[st
 	return out
 }
 
+// Values converts the values of a map to an array.
+func Values[T any](m map[string]T) []T {
+	var values []T
+	for _, version := range m {
+		values = append(values, version)
+	}
+	return values
+}
+
+// MapOf converts the values of an array to a map using a key function.
+func MapOf[T any](arr []T, keyFunc func(T) string) map[string]T {
```

Comment:
```suggestion
func CreateMapFromSlice[T any](arr []T, keyFunc func(T) string) map[string]T {
```

Nit.

Also, please add tests.